From abc7809628fb201f32da2069158305540b64e5c8 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Sat, 26 Apr 2014 10:57:05 -0500
Subject: [PATCH 38/38] PXE: distros implementing syslinux will be using raw
 kernels and initramfs

Since we are not detecting if the image is raw or wrapped and we expect
distros to be using raw images attempt bootz before bootm. This prevents
the user from getting a confusing error message:

Wrong Image Format for sysboot command
ERROR: can't get kernel image!
---
 common/cmd_pxe.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common/cmd_pxe.c b/common/cmd_pxe.c
index 18e140d..250243a 100644
--- a/common/cmd_pxe.c
+++ b/common/cmd_pxe.c
@@ -793,12 +793,12 @@ static int label_boot(cmd_tbl_t *cmdtp, struct pxe_label *label)
 	if (bootm_argv[3])
 		bootm_argc = 4;
 
-	do_bootm(cmdtp, 0, bootm_argc, bootm_argv);
-
 #ifdef CONFIG_CMD_BOOTZ
 	/* Try booting a zImage if do_bootm returns */
 	do_bootz(cmdtp, 0, bootm_argc, bootm_argv);
 #endif
+
+	do_bootm(cmdtp, 0, bootm_argc, bootm_argv);
 	return 1;
 }
 
-- 
1.9.0

